File: Tests/TestData/MultiFileCharacterization/SourceFiles/VisualBasicUsesCSharpRefReturn/ByRefArgument.vb
Public Class ByRefArgument
    Sub UseArr()
        Dim arrObj() As Object
        Modify(arrObj(0))

        Dim arrInt() As Integer
        Modify(arrInt(0))
    End Sub

    Sub UseRefReturn()
        Dim lstObj As CSharpRefReturn.RefReturnList(Of Object)
        Modify(lstObj(0))
        Modify(lstObj.RefProperty)

        Dim lstInt As CSharpRefReturn.RefReturnList(Of Integer)
        Modify(lstInt(0))
        Modify(lstInt.RefProperty)
    End Sub

    Sub Modify(ByRef o As Object)
    End Sub
End Class

File: Tests/TestData/MultiFileCharacterization/VBToCSResults/ConvertVbUsingCSharpRefReturnOnly/VisualBasicUsesCSharpRefReturn/ByRefArgument.cs
using Microsoft.VisualBasic.CompilerServices;

namespace VisualBasicUsesCSharpRefReturn
{
    public class ByRefArgument
    {
        public void UseArr()
        {
            var arrObj = default(object[]);
            Modify(ref arrObj[0]);

            var arrInt = default(int[]);
            var tmp = arrInt;
            object argo = tmp[0];
            Modify(ref argo);
            tmp[0] = Conversions.ToInteger(argo);
        }

        public void UseRefReturn()
        {
            var lstObj = default(CSharpRefReturn.RefReturnList<object>);
            Modify(ref lstObj[0]);
            Modify(ref lstObj.RefProperty);

            var lstInt = default(CSharpRefReturn.RefReturnList<int>);
            var tmp = lstInt;
            object argo = tmp[0];
            Modify(ref argo);
            tmp[0] = Conversions.ToInteger(argo);
            object argo1 = lstInt.RefProperty;
            Modify(ref argo1);
            lstInt.RefProperty = Conversions.ToInteger(argo1);
        }

        public void Modify(ref object o)
        {
        }
    }
}